rplugin/python3/deoplete/sources/deoplete_solargraph/core.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+import os
+import re
+import json
+import subprocess
+import signal
+from urllib.error import HTTPError
+
+from . import helpers
+from .helpers import *
+
+class Server:
+    def __init__(self, command='solargraph', args=['--port', '0']):
```

Comment:
Careful with mutable default arguments!
http://docs.python-guide.org/en/latest/writing/gotchas/

Reply:
This section is copy-pasted from unplugs/solargraph_utils.py

Reply:
Sry, it’s https://github.com/uplus/solargraph-utils.py